### PR TITLE
Check proxyAgent.options.path

### PR DESCRIPTION
**Pull request by zeptio**
Fixes bug where node http client where connecting using the proxy agent, even if it was not set.

Fixes: #93 

Signed-off-by: tobias <tobias@zept.io>

### CODE REVIEW

**Comment by thjaeckle** [comment]
Thanks you very much for the PR 👍 
Could you please add you to the NOTICE.md "Copyright Holders" as you did in the closed PR?

File: javascript/lib/node/src/node-http.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export class NodeRequester implements HttpRequester {
       host: parsedUrl.host,
       path: parsedUrl.pathname
     };
-    if (this.proxyAgent !== undefined) {
+    if (this.proxyAgent.options.path !== undefined) {
```
Comment (thjaeckle) [inline javascript/lib/node/src/node-http.ts:99]:
```suggestion
    if (this.proxyAgent !== undefined && this.proxyAgent.options.path !== undefined) {
```

Maybe adjust it like this in order to don't break what the creator of this intended?

**Comment by zeptio** [comment]
I've added the legal lotice. I'm still working on the websocket fix. So this is not ready yet.

File: javascript/lib/node/src/node-http.ts (hunk 1 above)
Comment (zeptio) [reply javascript/lib/node/src/node-http.ts:99]: The way it's written right now, this.proxyAgent will never be undefined. But sure, I can keep that part in.

**Comment by zeptio** [comment]
Now it's functional for both http and websocket.

**Review by thjaeckle — APPROVED** [review]
LGTM, thanks 👍 
(the reason we didn't find that earlier is because we always have to work behind a corporate proxy :/ ..)

**Pull request merged**